it/v1/endpoints/CreateAndAmendEmploymentExpensesControllerISpec.scala

```diff
@@ -294,9 +178,109 @@ class CreateAndAmendEmploymentExpensesControllerISpec extends IntegrationBaseSpe
+
+    def taxYear: String
+
+    val amount: BigDecimal = 123.12
+
+    def requestBodyJson: JsValue = Json.parse(
+      s"""
+         |{
+         |    "expenses": {
+         |        "businessTravelCosts": $amount,
+         |        "jobExpenses": $amount,
+         |        "flatRateJobExpenses": $amount,
+         |        "professionalSubscriptions": $amount,
+         |        "hotelAndMealExpenses": $amount,
+         |        "otherAndCapitalAllowances": $amount,
+         |        "vehicleExpenses": $amount,
+         |        "mileageAllowanceRelief": $amount
+         |    }
+         |}
+         |""".stripMargin
+    )
+
+    val hateoasResponse: JsValue = Json.parse(s"""
+                                                 |{
+                                                 |  "links": [
+                                                 |    {
+                                                 |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+                                                 |      "method": "GET",
+                                                 |      "rel": "self"
+                                                 |    },
+                                                 |    {
+                                                 |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+                                                 |      "method": "PUT",
+                                                 |      "rel": "amend-employment-expenses"
+                                                 |    },
+                                                 |    {
+                                                 |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+                                                 |      "method": "DELETE",
+                                                 |      "rel": "delete-employment-expenses"
+                                                 |    }
+                                                 |  ]
+                                                 |}
+                                                 |""".stripMargin)
+
+    def uri: String = s"/employments/$nino/$taxYear"
+
+    def downstreamUri: String
+
+    def setupStubs(): Unit = ()
+
+    def request(): WSRequest = {
+      AuditStub.audit()
+      AuthStub.authorised()
+      MtdIdLookupStub.ninoFound(nino)
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders(
+          (ACCEPT, "application/vnd.hmrc.1.0+json"),
+          (AUTHORIZATION, "Bearer 123") // some bearer token
+        )
+    }
+
+    def errorBody(code: String): String =
+      s"""
+         |{
+         |  "failures": [
+         |    {
+         |      "code": "$code",
+         |      "reason": "downstream message"
+         |    }
+         |  ]
+         |}
+    """.stripMargin
+
+  }
+
+  private trait NonTysTest extends Test {
+
+    def taxYear: String = "2021-22"
```

Comment:
these implementations (taxYear & downstreamUri) can all be vals without the String definition